fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/JoinPredicatePushdown.java

```diff
@@ -324,12 +325,15 @@ private void deriveIsNotNullPredicate(
         }
 
         LogicalJoinOperator joinOp = ((LogicalJoinOperator) join.getOp());
-        JoinOperator joinType = joinOp.getJoinType();
-        if ((joinType.isInnerJoin() || joinType.isRightSemiJoin()) && leftPushDown.isEmpty()) {
-            leftEQ.stream().map(c -> new IsNullPredicateOperator(true, c.clone(), true)).forEach(leftPushDown::add);
-        }
-        if ((joinType.isInnerJoin() || joinType.isLeftSemiJoin()) && rightPushDown.isEmpty()) {
-            rightEQ.stream().map(c -> new IsNullPredicateOperator(true, c.clone(), true)).forEach(rightPushDown::add);
+
+        if (ConnectContext.get().getSessionVariable().isCboDeriveJoinIsNullPredicate()) {
```

Comment:
why not return in the start of method?

Reply:
Yes. we can return at beginning of this method.

I'll fixed that.